### PR TITLE
🌱 Collect pods logs from workload clusters in E2E tests

### PR DESCRIPTION
**Pull request by tcordeu**
**What this PR does / why we need it**:
Collect pods logs from workload clusters in E2E tests.

**Which issue(s) this PR fixes**
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/3641

### CODE REVIEW

**Review by fabriziopandini — COMMENTED** [review]
I checked the output of this change in the [PR job](https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api/4017/pull-cluster-api-e2e-main/1338940946871488512/artifacts/clusters/quick-start-xv7cga/machines/quick-start-xv7cga-control-plane-848xz/pods/) and it looks great!
/lgtm

File: test/framework/docker_logcollector.go [hunk 1]
```diff
@@ -54,6 +56,35 @@ func (k DockerLogCollector) CollectMachineLog(ctx context.Context, managementClu
...
+
+			tempfileName := f.Name()
+			outputDir := filepath.Join(outputPath, dirName)
+
+			defer os.Remove(tempfileName)
+
+			err = execOnContainer(
+				containerName,
+				f,
+				"tar", "--hard-dereference", "-C", containerDir, "-chf", "-", ".",
```
Comment (wfernandes) [inline test/framework/docker_logcollector.go:74]: nit:
Just for verbosity and for those that may not know the flags of tar, suggest expanding these to their full flag names. 🙂 

```suggestion
				"tar", "--hard-dereference", "--dereference", "--directory", containerDir,  "--create", "--file", "-", ".",
```

**Review by wfernandes — COMMENTED** [review]
This is great! Thanks for this PR. Just one minor nit/suggestion.

File: test/framework/docker_logcollector.go (hunk 1 above)
Comment (tcordeu) [reply test/framework/docker_logcollector.go]: Done, expanded the flags on both commands.

**Comment by tcordeu** [comment]
@wfernandes @fabriziopandini could someone trigger a retest of `pull-cluster-api-test-main`, seems to be flaky. Unless I am missing something.

**Comment by wfernandes** [comment]
@tcordeu I believe members should be able to kick of tests as well. 
/test pull-cluster-api-test-main

**Pull request merged**